Update context menu images for internal clients

lilyspiniolas opened the pull request:
#### a025bc40ba4754766f3949c823bb88a3c42bc029
<pre>
Update context menu images for internal clients
<a href="https://bugs.webkit.org/show_bug.cgi?id=291707">https://bugs.webkit.org/show_bug.cgi?id=291707</a>
<a href="https://rdar.apple.com/149504756">rdar://149504756</a>

Reviewed by Abrar Rahman Protyasha.

Updated context menu images for internal clients
where needed, including removing unneeded ones.

The UnifiedPDF item which opened the PDF in an external
viewer now displays the name of the default viewer rather
than always displaying &quot;Preview&quot;. When context menu images
for internal clients is enabled, the image for this item
is set to the icon of the default PDF viewer.

In the &quot;Font&quot; submenu, the separator between &quot;Styles...&quot; and
&quot;Show Color&quot; has been moved above the &quot;Outline&quot; item, and
a new separator has been placed above the &quot;Bold&quot; item. This
results in a visually distinct section for the 3 checkable items
(Bold, Italic, Underline).

* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::createAndAppendFontSubMenu):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::contextMenuItemPDFOpenWithDefaultViewer):
(WebCore::contextMenuItemPDFOpenWithPreview): Deleted.
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebKit/Platform/mac/MenuUtilities.mm:
(WebKit::symbolNameForAction):
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(WebKit::WebContextMenuProxyMac::createShareMenuItem):
(WebKit::WebContextMenuProxyMac::getContextMenuItem):
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::showPDFContextMenu):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::contextMenuActionFromTag const):
(WebKit::UnifiedPDFPlugin::toContextMenuItemTag):
(WebKit::UnifiedPDFPlugin::createContextMenu const):
(WebKit::UnifiedPDFPlugin::titleForContextMenuItemTag const):
(WebKit::UnifiedPDFPlugin::contextMenuItem const):
(WebKit::UnifiedPDFPlugin::performContextMenuAction):

Canonical link: <a href="https://commits.webkit.org/293840@main">https://commits.webkit.org/293840@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c590997df1ad57d8729c94ceea53cc1d505da5c9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100048 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19696 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9989 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105200 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50629 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102089 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20002 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28169 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/76173 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33232 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103055 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15275 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90358 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56534 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15087 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8353 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49998 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85003 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8438 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107536 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27161 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19892 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85112 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27524 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86563 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84644 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21506 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29317 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7055 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20997 "Built successfully") | 
| [  ~~🛠 🧪 unsafe-merge~~](https://ews-build.webkit.org/#/builders/22/builds/16282 "The change is no longer eligible for processing.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27098 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32327 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26909 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30225 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28468 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
```diff
@@ -589,18 +589,38 @@ static inline bool expectsLegacyImplicitRubberBandControl()
+
         RetainPtr nsItem = adoptNS([[NSMenuItem alloc] init]);
-        [nsItem setTitle:item.title.createNSString().get()];
+        RetainPtr<String> defaultPDFViewerPath;
```
This seems wrong

aprotyas replied:
Yeah I would remove this line

aprotyas commented on Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
```diff
@@ -589,18 +589,37 @@ static inline bool expectsLegacyImplicitRubberBandControl()
         RetainPtr nsItem = adoptNS([[NSMenuItem alloc] init]);
-        [nsItem setTitle:item.title.createNSString().get()];
+
+#if ENABLE(CONTEXT_MENU_IMAGES_FOR_INTERNAL_CLIENTS)
+        auto shouldSetImage = m_preferences->contextMenuImagesForInternalClientsEnabled();
+#endif
```
Nit: If you define `shouldSetImage` like this, you won't need to use the platform guard in multiple places. Optional though.

```suggestion
    auto shouldSetImage = [preferences = m_preferences] {
#if ENABLE(CONTEXT_MENU_IMAGES_FOR_INTERNAL_CLIENTS)
        return preferences->contextMenuImagesForInternalClientsEnabled();
#else
        return false;
#endif
    }();
```

lilyspiniolas committed:
Update context menu images for internal clients
https://bugs.webkit.org/show_bug.cgi?id=291707
rdar://149504756

Reviewed by Abrar Rahman Protyasha.

Updated context menu images for internal clients
where needed, including removing unneeded ones.

The UnifiedPDF item which opened the PDF in an external
viewer now displays the name of the default viewer rather
than always displaying "Preview". When context menu images
for internal clients is enabled, the image for this item
is set to the icon of the default PDF viewer.

In the "Font" submenu, the separator between "Styles..." and
"Show Color" has been moved above the "Outline" item, and
a new separator has been placed above the "Bold" item. This
results in a visually distinct section for the 3 checkable items
(Bold, Italic, Underline).

* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::createAndAppendFontSubMenu):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::contextMenuItemPDFOpenWithDefaultViewer):
(WebCore::contextMenuItemPDFOpenWithPreview): Deleted.
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebKit/Platform/mac/MenuUtilities.mm:
(WebKit::symbolNameForAction):
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(WebKit::WebContextMenuProxyMac::createShareMenuItem):
(WebKit::WebContextMenuProxyMac::getContextMenuItem):
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::showPDFContextMenu):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::contextMenuActionFromTag const):
(WebKit::UnifiedPDFPlugin::toContextMenuItemTag):
(WebKit::UnifiedPDFPlugin::createContextMenu const):
(WebKit::UnifiedPDFPlugin::titleForContextMenuItemTag const):
(WebKit::UnifiedPDFPlugin::contextMenuItem const):
(WebKit::UnifiedPDFPlugin::performContextMenuAction):

Canonical link: https://commits.webkit.org/293840@main
